lib/Canonicalize/MakeModelGEPPass.cpp

```diff
@@ -1831,46 +1764,40 @@ getAccessedTypeOnIR(FunctionMetadataCache &Cache,
       }
 
     } else if (isCallToIsolatedFunction(Call)) {
-      auto *ProtoT = Cache.getCallSitePrototype(Model, Call).get();
+      auto *ProtoT = Cache.getCallSitePrototype(Model, Call);
       revng_assert(ProtoT != nullptr);
 
-      if (const auto *RFT = dyn_cast<RawFunctionDefinition>(ProtoT)) {
+      if (const auto *RFT = llvm::dyn_cast<RawFunctionDefinition>(ProtoT)) {
 
         auto MoreIndent = LoggerIndent(ModelGEPLog);
         auto ModelArgSize = RFT->Arguments().size();
-        auto &Type = RFT->StackArgumentsType();
-        revng_assert((ModelArgSize == Call->arg_size() - 1 and Type.isValid())
+        bool HasStackArgs = not RFT->StackArgumentsType().empty();
+        revng_assert((ModelArgSize == Call->arg_size() - 1 and HasStackArgs)
                      or ModelArgSize == Call->arg_size());
         revng_log(ModelGEPLog, "model::RawFunctionDefinition");
 
         auto _ = LoggerIndent(ModelGEPLog);
         if (not Call->isCallee(&U)) {
-          unsigned ArgOpNum = Call->getArgOperandNo(&U);
-          revng_log(ModelGEPLog, "ArgOpNum: " << ArgOpNum);
+          unsigned N = Call->getArgOperandNo(&U);
+          revng_log(ModelGEPLog, "ArgOpNum: " << N);
           revng_log(ModelGEPLog, "ArgOperand: " << U.get());
 
-          model::QualifiedType ArgTy;
-          if (ArgOpNum >= ModelArgSize) {
-            // The only case in which the argument's index can be greater than
-            // the number of arguments in the model is for RawFunctionDefinition
-            // functions that have stack arguments.
-            // Stack arguments are passed as the last argument of the llvm
-            // function, but they do not have a corresponding argument in the
-            // model. In this case, we have to retrieve the StackArgumentsType
-            // from the function prototype.
-            ArgTy = { RFT->StackArgumentsType(), {} };
-            revng_assert(ArgTy.UnqualifiedType().isValid());
-          } else {
-            auto ArgIt = std::next(RFT->Arguments().begin(), ArgOpNum);
-            ArgTy = ArgIt->Type();
-          }
-
-          revng_log(ModelGEPLog,
-                    "model::QualifiedType: " << serializeToString(ArgTy));
-          if (ArgTy.isPointer()) {
-            model::QualifiedType Pointee = dropPointer(ArgTy);
-            revng_log(ModelGEPLog, "Pointee: " << serializeToString(Pointee));
-            return Pointee;
+          // The only case in which the RawFunctionDefinition's argument index
+          // can be greater than the number of arguments in the model is when
```

Comment:
"...greater than the number of *register* arguments..."

Reply:
Done.